prompttools/experiment/experiments/_utils.py

```diff
@@ -41,4 +41,9 @@ def _get_dynamic_columns(df: pd.DataFrame) -> pd.DataFrame:
 
     unique_counts = df[hashable_columns].nunique()
     columns_with_multiple_unique_values = unique_counts[unique_counts > 1].index
-    return pd.concat([df[columns_with_multiple_unique_values], df[unhashable_columns]], axis=1)
+    dfs_to_concat = [df[columns_with_multiple_unique_values], df[unhashable_columns]]
+    if 'prompt' in df:
+        dfs_to_concat.append(df['prompt'])
+    if 'messages' in df:
+        dfs_to_concat.append(df['messages'])
```

Comment:
Might need to check if it already exists in `dfs_to_concat` . Otherwise, you might get duplicate? Unless `pd.concat` handles duplication

Reply:
Updated the logic to avoid duplicates